src/core/transaction.zig

```diff
@@ -360,3 +451,117 @@ test "core.transaction: Message sanitize fails if account index is out of bounds
+        184, 2,   85,  213, 101, 44,  13,  181, 13,  65,  128, 17,  94,  229, 31,  215, 47,  49,
+        72,  57,  158, 144, 193, 224, 205, 241, 120, 78,  5,   1,   3,   5,   7,   90,  0,
+    };
+};
+
+pub const test_v0_message = struct {
+    pub fn asStruct(allocator: std.mem.Allocator) !V0Message {
+        return V0Message{
+            .header = MessageHeader{
+                .num_required_signatures = 39,
+                .num_readonly_signed_accounts = 12,
+                .num_readonly_unsigned_accounts = 102,
+            },
+            .account_keys = try allocopy(allocator, Pubkey, &.{
+                try Pubkey.fromString("GubTBrbgk9JwkwX1FkXvsrF1UC2AP7iTgg8SGtgH14QE"),
+                try Pubkey.fromString("5yCD7QeAk5uAduhLZGxePv21RLsVEktPqJG5pbmZx4J4"),
+            }),
+            .recent_blockhash = try Hash
+                .parseBase58String("4xzjBNLkRqhBVmZ7JKcX2UEP8wzYKYWpXk7CPXzgrEZW"),
+            .instructions = try allocopy(allocator, CompiledInstruction, &.{.{
+                .program_id_index = 100,
+                .accounts = try allocopy(allocator, u8, &.{ 1, 3 }),
+                .data = try allocopy(allocator, u8, &.{
+                    104, 232, 42,  254, 46, 48, 104, 89,  101, 211, 253, 161, 65, 155, 204, 89,
+                    126, 187, 180, 191, 60, 59, 88,  119, 106, 20,  194, 80,  11, 200, 76,  0,
+                }),
+            }}),
+            .address_table_lookups = try allocopy(allocator, MessageAddressTableLookup, &.{.{
+                .account_key = try Pubkey.fromString("ZETAxsqBRek56DhiGXrn75yj2NHU3aYUnxvHXpkf3aD"),
+                .writable_indexes = try allocopy(allocator, u8, &.{ 1, 3, 5, 7, 90 }),
+                .readonly_indexes = try allocopy(allocator, u8, &.{}),
+            }}),
+        };
+    }
+
+    pub const bincode_serialized_bytes = [_]u8{
+        39,  12,  102, 2,   236, 88,  117, 221, 34,  125, 55,  183, 193, 174, 21,  99,  70,  167,
+        52,  227, 254, 241, 14,  239, 13,  172, 158, 81,  254, 134, 30,  78,  35,  15,  168, 79,
+        73,  211, 242, 100, 122, 21,  163, 216, 62,  58,  230, 205, 163, 112, 95,  100, 134, 113,
+        98,  129, 164, 240, 184, 157, 4,   34,  55,  72,  89,  113, 179, 97,  58,  235, 71,  20,
+        83,  42,  196, 46,  189, 136, 194, 90,  249, 14,  154, 144, 141, 234, 253, 148, 146, 168,
+        110, 10,  237, 82,  157, 190, 248, 20,  215, 105, 1,   100, 2,   1,   3,   32,  104, 232,
+        42,  254, 46,  48,  104, 89,  101, 211, 253, 161, 65,  155, 204, 89,  126, 187, 180, 191,
+        60,  59,  88,  119, 106, 20,  194, 80,  11,  200, 76,  0,   1,   8,   65,  203, 149, 184,
+        2,   85,  213, 101, 44,  13,  181, 13,  65,  128, 17,  94,  229, 31,  215, 47,  49,  72,
+        57,  158, 144, 193, 224, 205, 241, 120, 78,  5,   1,   3,   5,   7,   90,  0,
+    };
+};
+
+fn allocopy(allocator: std.mem.Allocator, comptime T: type, source: []const T) ![]T {
```

Comment:
Is this different to `allocator.dupe(comptime T: type, source: []const T)`

Reply:
ec46d34e5a63dcc4218b8fa67c162e63555c0da0